### PR TITLE
Use new eKuiper security flag, subscribe to ASC

### PR DESCRIPTION
**Pull request by farshidtz**
Depends on https://github.com/canonical/edgex-ekuiper-snap/pull/34

### CODE REVIEW

**Comment by farshidtz** [comment]
Passes locally using the snap build from https://github.com/canonical/edgex-ekuiper-snap/pull/34, by replacing the channel to `edge/pr-34-review`.

```
$ go test -v ./test/suites/edgex-no-sec --run=TestRulesEngine
2022/10/24 15:33:37 [CLEAN]
2022/10/24 15:33:37 [exec] sudo snap remove --purge edgexfoundry
2022/10/24 15:33:37 [stderr] snap "edgexfoundry" is not installed
2022/10/24 15:33:37 [exec] sudo snap remove --purge edgex-device-virtual
2022/10/24 15:33:37 [stderr] snap "edgex-device-virtual" is not installed
2022/10/24 15:33:37 [exec] sudo snap remove --purge edgex-ekuiper
2022/10/24 15:33:37 [stderr] snap "edgex-ekuiper" is not installed
2022/10/24 15:33:37 [exec] sudo snap remove --purge edgex-app-service-configurable
2022/10/24 15:33:37 [stderr] snap "edgex-app-service-configurable" is not installed
2022/10/24 15:33:37 [SETUP]
2022/10/24 15:33:37 [exec] sudo snap install edgexfoundry --channel=latest/edge
2022/10/24 15:34:43 [stdout] edgexfoundry (edge) 2.3.0-dev.99 from Canonical** installed
2022/10/24 15:34:43 [exec] sudo snap install edgex-device-virtual --channel=latest/edge
2022/10/24 15:34:52 [stdout] edgex-device-virtual (edge) 2.3.0-dev.25 from Canonical** installed
2022/10/24 15:34:52 [exec] sudo snap install edgex-ekuiper --channel=edge/pr-34-review
2022/10/24 15:35:03 [stdout] edgex-ekuiper (edge/pr-34-review) 1.5.1+snap+git24.1a20ffa from Canonical** installed
2022/10/24 15:35:03 [exec] sudo snap install edgex-app-service-configurable --channel=latest/edge
2022/10/24 15:35:12 [stdout] edgex-app-service-configurable (edge) 2.3.0-dev.31 from Canonical** installed
2022/10/24 15:35:12 [exec] sudo snap set edgexfoundry security-secret-store='off'
2022/10/24 15:35:27 [stdout] 2022-10-24T15:35:27+02:00 INFO task ignored
2022/10/24 15:35:27 [exec] sudo snap set edgex-device-virtual config.edgex-security-secret-store='false'
2022/10/24 15:35:27 [exec] sudo snap set edgex-app-service-configurable app-options='true'
2022/10/24 15:35:27 [exec] sudo snap set edgex-app-service-configurable config.edgex-security-secret-store='false'
2022/10/24 15:35:28 [exec] sudo snap set edgex-ekuiper config.edgex-security-secret-store='false'
2022/10/24 15:35:28 [exec] sudo snap start edgex-device-virtual
2022/10/24 15:35:28 [stdout] Started.
2022/10/24 15:35:28 Retry 1/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:29 Retry 2/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:30 Retry 3/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:31 Retry 4/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:32 Retry 5/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:33 Retry 6/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:34 Retry 7/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:35 Retry 8/60: Waiting for ports: 59900 (device-virtual)
2022/10/24 15:35:35 [exec] sudo snap set edgex-ekuiper snap set edgex-ekuiper config.edgex.default.topic='rules-events'
2022/10/24 15:35:36 [stderr] error: invalid configuration: "snap" (want key=value)
2022/10/24 15:35:36 [exec] sudo snap set edgex-ekuiper snap set edgex-ekuiper config.edgex.default.messagetype='event'
2022/10/24 15:35:36 [stderr] error: invalid configuration: "snap" (want key=value)
2022/10/24 15:35:36 [exec] sudo snap start edgex-ekuiper
2022/10/24 15:35:36 [stdout] Started.
2022/10/24 15:35:36 Retry 1/60: Waiting for ports: 20498 (ekuiper), 59720 (ekuiper/rest-api)
2022/10/24 15:35:37 Retry 2/60: Waiting for ports: 20498 (ekuiper), 59720 (ekuiper/rest-api)
2022/10/24 15:35:37 [exec] sudo snap set edgex-app-service-configurable profile='rules-engine'
2022/10/24 15:35:37 [exec] sudo snap start edgex-app-service-configurable
2022/10/24 15:35:38 [stdout] Started.
2022/10/24 15:35:38 Retry 1/60: Waiting for ports: 59701 (app-service-configurable)
2022/10/24 15:35:39 Retry 2/60: Waiting for ports: 59701 (app-service-configurable)
2022/10/24 15:35:39 Retry 1/180: Waiting for ports: 59880 (core-data), 59881 (core-metadata), 59882 (core-command), 8500 (consul), 6379 (redis)
=== RUN   TestRulesEngine
    exec.go:19: [exec] sudo snap start edgex-ekuiper.kuiper
    exec.go:93: [stdout] Started.
    exec.go:19: [exec] sudo snap start edgex-device-virtual
    exec.go:93: [stdout] Started.
    exec.go:19: [exec] sudo snap start edgex-app-service-configurable
    exec.go:93: [stdout] Started.
=== RUN   TestRulesEngine/create_stream_and_rule
    exec.go:19: [exec] edgex-ekuiper.kuiper-cli create stream stream1 '()WITH(FORMAT="JSON",TYPE="edgex")'
    exec.go:19: [exec] edgex-ekuiper.kuiper-cli create rule rule_edgex_message_bus '
                                {
                                   "sql":"SELECT * from stream1",
                                   "actions": [
                                          {
                                                 "edgex": {
                                                        "connectionSelector": "edgex.redisMsgBus",
                                                        "topicPrefix": "edgex/events/device", 
                                                        "messageType": "request",
                                                        "deviceName": "device-test"
                                                 }
                                          }
                                   ]
                                }'
    net.go:140: Retry 1/60: Waiting for ports: 59900 (device-virtual)
=== RUN   TestRulesEngine/create_stream_and_rule/query_readings
    readings.go:45: waiting for device-virtual produce readings, current retry count: 1/60
    readings.go:45: waiting for device-virtual produce readings, current retry count: 2/60
    readings.go:45: waiting for device-virtual produce readings, current retry count: 3/60
    readings.go:45: waiting for device-virtual produce readings, current retry count: 4/60
    readings.go:45: waiting for device-virtual produce readings, current retry count: 5/60
    readings.go:45: waiting for device-virtual produce readings, current retry count: 6/60
    readings.go:45: waiting for device-virtual produce readings, current retry count: 7/60
    readings.go:48: device-virtual is producing readings now, readings queried from core-data
=== CONT  TestRulesEngine
    exec.go:19: [exec] sudo snap stop edgex-ekuiper.kuiper
    exec.go:93: [stdout] Stopped.
    exec.go:19: [exec] sudo snap stop edgex-device-virtual
    exec.go:93: [stdout] Stopped.
    exec.go:19: [exec] sudo snap stop edgex-app-service-configurable
    exec.go:93: [stdout] Stopped.
--- PASS: TestRulesEngine (8.94s)
    --- PASS: TestRulesEngine/create_stream_and_rule (7.20s)
        --- PASS: TestRulesEngine/create_stream_and_rule/query_readings (7.01s)
PASS
2022/10/24 15:35:47 [TEARDOWN]
2022/10/24 15:35:47 [exec] (sudo journalctl --since "2022-10-24 15:33:37" --no-pager | grep "edgexfoundry"|| true) > edgexfoundry.log
Wrote snap logs to /home/farshid/projects/edgex/edgex-snap-testing/test/suites/edgex-no-sec/edgexfoundry.log
2022/10/24 15:35:48 [exec] (sudo journalctl --since "2022-10-24 15:33:37" --no-pager | grep "edgex-device-virtual"|| true) > edgex-device-virtual.log
Wrote snap logs to /home/farshid/projects/edgex/edgex-snap-testing/test/suites/edgex-no-sec/edgex-device-virtual.log
2022/10/24 15:35:48 [exec] (sudo journalctl --since "2022-10-24 15:33:37" --no-pager | grep "edgex-ekuiper"|| true) > edgex-ekuiper.log
Wrote snap logs to /home/farshid/projects/edgex/edgex-snap-testing/test/suites/edgex-no-sec/edgex-ekuiper.log
2022/10/24 15:35:48 [exec] (sudo journalctl --since "2022-10-24 15:33:37" --no-pager | grep "edgex-app-service-configurable"|| true) > edgex-app-service-configurable.log
Wrote snap logs to /home/farshid/projects/edgex/edgex-snap-testing/test/suites/edgex-no-sec/edgex-app-service-configurable.log
2022/10/24 15:35:48 [exec] sudo snap remove --purge edgexfoundry
2022/10/24 15:35:49 [stdout] 2022-10-24T15:35:49+02:00 INFO Waiting for "snap.edgexfoundry.core-data.service" to stop.
2022/10/24 15:35:50 [stdout] 2022-10-24T15:35:50+02:00 INFO Waiting for "snap.edgexfoundry.core-metadata.service" to stop.
2022/10/24 15:36:11 [stdout] edgexfoundry removed
2022/10/24 15:36:11 [exec] sudo snap remove --purge edgex-device-virtual
2022/10/24 15:36:15 [stdout] edgex-device-virtual removed
2022/10/24 15:36:15 [exec] sudo snap remove --purge edgex-ekuiper
2022/10/24 15:37:12 [stdout] edgex-ekuiper removed
2022/10/24 15:37:12 [exec] sudo snap remove --purge edgex-app-service-configurable
2022/10/24 15:37:15 [stdout] edgex-app-service-configurable removed
ok      edgex-snap-testing/test/suites/edgex-no-sec     217.921s
```

File: test/suites/edgex-no-sec/main_test.go [hunk 1]
```diff
@@ -116,7 +116,10 @@ func setup() (teardown func(), err error) {
 		return
 	}
 
-	utils.SnapRestart(nil, ekuiperSnap)
+	// subscribe to ASC events
+	utils.SnapSet(nil, ekuiperSnap, "snap set edgex-ekuiper config.edgex.default.topic", "rules-events")
+	utils.SnapSet(nil, ekuiperSnap, "snap set edgex-ekuiper config.edgex.default.messagetype", "event")
```
Comment (MonicaisHer) [inline test/suites/edgex-no-sec/main_test.go:121]:
```suggestion
	utils.SnapSet(nil, ekuiperSnap, "config.edgex.default.topic", "rules-events")
	utils.SnapSet(nil, ekuiperSnap, "config.edgex.default.messagetype", "event")
```

**Comment by farshidtz** [comment]
The edgex-no-sec tests are passing now.

**Review by MonicaisHer — APPROVED** [review]
@farshidtz Thank you, looks good.

**Pull request merged**